chia/full_node/mempool_manager.py

```diff
@@ -614,24 +612,19 @@ async def new_peak(
         if use_optimization and last_npc_result is not None:
             # We don't reinitialize a mempool, just kick removed items
             if last_npc_result.conds is not None:
-                spendbundle_ids_to_remove = []
+                spendbundle_ids_to_remove: List[bytes32] = []
                 for spend in last_npc_result.conds.spends:
-                    if spend.coin_id in self.mempool.removal_coin_id_to_spendbundle_ids:
-                        spendbundle_ids: List[bytes32] = self.mempool.removal_coin_id_to_spendbundle_ids[
-                            bytes32(spend.coin_id)
-                        ]
-                        spendbundle_ids_to_remove.extend(spendbundle_ids)
-                        for spendbundle_id in spendbundle_ids:
-                            item = self.mempool.spends.get(spendbundle_id)
-                            if item:
-                                included_items.append(item)
-                            self.remove_seen(spendbundle_id)
-                self.mempool.remove_from_pool(spendbundle_ids_to_remove, MempoolRemoveReason.BLOCK_INCLUSION)
+                    items: List[MempoolItem] = self.mempool.get_spends_by_coin_id(bytes32(spend.coin_id))
+                    for item in items:
+                        included_items.append(item)
+                        self.remove_seen(item.name)
+                        spendbundle_ids_to_remove.append(item.name)
+                    self.mempool.remove_from_pool(spendbundle_ids_to_remove, MempoolRemoveReason.BLOCK_INCLUSION)
```

Comment:
This one got altered from being called once in the end, to being called several times. I doubt it's intentional?

Reply:
last minute merge conflict..